run_macro_chunking_experiments.py

```diff
@@ -0,0 +1,209 @@
+import click
+import torch.cuda
+from mteb import MTEB
+from transformers import AutoModel, AutoTokenizer
+
+from chunked_pooling.chunked_eval_tasks import (
+    LEMBWikimQARetrievalChunked, 
+    LEMBNeedleRetrievalChunked, 
+    LEMBNarrativeQARetrievalChunked,
+    LEMBQMSumRetrievalChunked,
+    LEMBSummScreenFDRetrievalChunked
+)
+from chunked_pooling.wrappers import load_model
+
+DEFAULT_CHUNKING_STRATEGY = 'fixed'
+DEFAULT_CHUNK_SIZE = 256
+DEFAULT_N_SENTENCES = 5
+BATCH_SIZE = 1
+DEFAULT_SOFT_BOUNDARY_EMBED_SIZE = 8192
+DEFAULT_HARD_BOUNDARY_EMBED_SIZE = 0
+
+
+
+@click.command()
+@click.option(
+    '--model-name',
+    default='jinaai/jina-embeddings-v2-small-en',
+    help='The name of the model to use.',
+)
+@click.option(
+    '--strategy',
+    default=DEFAULT_CHUNKING_STRATEGY,
+    help='The chunking strategy to be applied.',
+)
+@click.option(
+    '--task-name', default='LEMBWikimQARetrievalChunked', help='The evaluation task to perform.'
+)
+@click.option(
+    '--eval-split', default='test', help='The name of the evaluation split in the task.'
+)
+@click.option(
+    '--chunking-model',
+    default=None,
+    required=False,
+    help='The name of the model used for semantic chunking.',
+)
+@click.option(
+    '--truncate-max-length',
+    default=8192,
+    type=int,
+    help='Maximum number of tokens; By default, no truncation is done.',
+)
+@click.option(
+    '--chunk-size',
+    default=DEFAULT_CHUNK_SIZE,
+    type=int,
+    help='Number of tokens per chunk for fixed strategy.',
+)
+@click.option(
+    '--n-sentences',
+    default=DEFAULT_N_SENTENCES,
+    type=int,
+    help='Number of sentences per chunk for sentence strategy.',
+)
+@click.option(
+    '--soft-boundary-embed-size',
+    default=DEFAULT_SOFT_BOUNDARY_EMBED_SIZE,
+    type=int,
+    help='Token length of the embeddings that come before/after soft boundaries (i.e. overlapping embeddings). Above zero, soft boundaries are used.',
+)
+@click.option(
+    '--hard-boundary-embed-size',
+    default=DEFAULT_HARD_BOUNDARY_EMBED_SIZE,
+    type=int,
+    help='Token length of the embeddings that come before/after hard boundaries. Above zero, hard boundaries are used.',
+)
+
+def main(
+    model_name,
+    strategy,
+    task_name,
+    eval_split,
+    chunking_model,
+    truncate_max_length,
+    chunk_size,
+    n_sentences,
+    soft_boundary_embed_size,
+    hard_boundary_embed_size,
+):
```

Comment:
As we didn't run this experiment, I think it does not make sense to merge this script